sympy/functions/elementary/complexes.py

```diff
@@ -863,6 +868,15 @@ class conjugate(DefinedFunction):
     """
     _singularities = True  # non-holomorphic
 
+    if TYPE_CHECKING:
+
+        @overload
+        def __new__(cls, arg: Quaternion) -> Quaternion: ... # type: ignore
+        @overload
+        def __new__(cls, arg: Expr) -> Expr: ... # type: ignore
+
+        def __new__(cls, arg: Basic) -> Basic: ... # type: ignore
+
```

Comment:
The type:ignore here are needed because of: https://github.com/python/mypy/issues/15182